sql/planbuilder/dateparse/date.go

```diff
@@ -11,6 +11,31 @@ var (
 	timeSpecifiers = []uint8{'f', 'H', 'h', 'I', 'i', 'k', 'l', 'p', 'r', 'S', 's', 'T'}
 )
 
+func HasDateOrTime(format string) (bool, bool, error) {
```

Comment:
Giving this function some godocs would help people more quickly understand it without having to read the implementation. Alternatively, naming the two boolean return params could also go a long way to making this function easier for people to use/understand.